synapse/tools/rstorm.py

```diff
@@ -18,17 +27,143 @@ async def cortex():
+    loc = s_dyndeps.getDynLocal(ctor)
+    svc = await loc.anit(dirn, conf=svcconf)
+    core.onfini(svc.fini)
+
+    svc.dmon.share('svc', svc)
+    root = await svc.auth.getUserByName('root')
+    await root.setPasswd('root')
+    info = await svc.dmon.listen('tcp://127.0.0.1:0/')
+    svc.dmon.test_addr = info
+    host, port = info
+    surl = f'tcp://root:root@127.0.0.1:{port}/svc'
+    await core.nodes(f'service.add {svcname} {surl}')
+    await core.nodes(f'$lib.service.wait({svcname})')
+
+
+class StormOutput(s_cmds_cortex.StormCmd):
     '''
     Produce standard output from a stream of storm runtime messages.
+    Must be instantiated for a single query with a rstorm context.
     '''
     # TODO: Eventually make this obey all cmdr options and swap it in
 
+    _cmd_syntax = (
+        ('--hide-query', {}),
+    ) + s_cmds_cortex.StormCmd._cmd_syntax
+
+    def __init__(self, core, ctx, stormopts=None, opts=None):
+        if opts is None:
+            opts = {}
+
+        s_cmds_cortex.StormCmd.__init__(self, None, **opts)
+
+        self.stormopts = stormopts or {}
+
+        # hide a few mesg types by default
+        for mtype in ('init', 'fini', 'node:edits', 'node:edits:count', 'prov:new'):
+            self.cmdmeths[mtype] = self._silence
+
+        self.core = core
+        self.ctx = ctx
+        self.lines = []
+
+    async def runCmdLine(self, line):
+        opts = self.getCmdOpts(f'storm {line}')
+        return await self.runCmdOpts(opts)
+
+    async def _mockHttp(self, *args, **kwargs):
+        info = {
+            'code': 200,
+            'body': '{}',
+        }
+
+        resp = self.ctx.get('mock-http')
+        if resp:
+            body = resp.get('body')
+
+            if isinstance(body, dict):
+                body = json.dumps(body)
+
+            info = {
+                'code': resp.get('code', 200),
+                'body': body,
+            }
+
+        return s_stormhttp.HttpResp(info)
+
+    def printf(self, mesg, addnl=True, color=None):
+        line = f'    {mesg}'
+        self.lines.append(line)
+        return line
+
+    def _silence(self, mesg, opts):
+        pass
+
+    def _onErr(self, mesg, opts):
+        # raise on err for rst
+        raise s_exc.StormRuntimeError(mesg=mesg)
+
+    async def runCmdOpts(self, opts):
+
+        text = opts.get('query')
+
+        if not opts.get('hide-query'):
+            self.printf(f'> {text}')
+
+        stormopts = copy.deepcopy(self.stormopts)
+
+        stormopts.setdefault('repr', True)
+        stormopts.setdefault('path', opts.get('path', False))
+
+        hide_unknown = True
+
+        # TODO
+        # showtext = opts.get('show')
+        # if showtext is not None:
+        #     stormopts['show'] = showtext.split(',')
+        #
+        # editformat = opts['editformat']
+        # if editformat != 'nodeedits':
+        #     stormopts['editformat'] = editformat
+
+        stormopts['editformat'] = 'nodeedits'
```

Comment:
Do we need edits for docs at all?

Reply:
Not so far, and I explicitly exclude from the printfs.  I removed that block.